metrics/web/test/common/presentation/navigation/state/navigation_notifier_test.dart

```diff
@@ -167,13 +168,91 @@ void main() {
       },
     );
 
+    test(
+      ".handleLoggedOut() clears pages when a user logs out",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(1));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() pushes to the login page when a user logs out",
+      () {
+        notifier.handleLoggedIn();
+        notifier.push(DefaultRoutes.dashboard);
+
+        notifier.handleLoggedOut();
+
+        final currentPage = notifier.pages.last;
+
+        expect(currentPage.name, isLoginPageName);
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not clear pages when a user logs out from the login page",
+      () {
+        notifier.push(DefaultRoutes.login);
+
+        notifier.handleLoggedOut();
+
+        expect(notifier.pages, hasLength(2));
+      },
+    );
+
+    test(
+      ".handleLoggedOut() does not push to the login page when a user logs out from the login page",
+      () {
+        notifier.handleLoggedOut();
+
+        final loginPagesCount = notifier.pages
```

Comment:
✅ 
Maybe we can just check that the length of the pages does not change? What do you think?